addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjection.java

```diff
@@ -0,0 +1,705 @@
+    private static final String STOP_LOG = "Stopping the scan due to a user request";
+    private static final Logger LOG = Logger.getLogger(MongoDbInjection.class);
+
+    @Override
+    public int getCweId() {
+        return 943;
+    }
+
+    @Override
+    public int getWascId() {
+        return 19;
+    }
+
+    public int getId() {
+        return 40033;
+    }
+
+    public Tech getTech() {
+        // TODO change in Tech.MongoDB as soon as available
+        return Tech.Db;
+    }
+
+    @Override
+    public boolean targets(TechSet technologies) {
+        // TODO change in Tech.MongoDB as soon as available
+        return technologies.includes(Tech.Db);
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "name");
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+    }
+
+    @Override
+    public int getCategory() {
+        return Category.INJECTION;
+    }
+
+    @Override
+    public String getSolution() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+    }
+
+    @Override
+    public String getReference() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+    }
+
+    @Override
+    public String[] getDependency() {
+        return new String[] {};
+    }
+
+    public String getExtraInfo(String attack) {
+        return Constant.messages.getString(MESSAGE_PREFIX + "extrainfo." + attack);
+    }
+
+    @Override
+    public void init() {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Initialising MongoDB penertration tests");
+        }
+        if (this.getAttackStrength() == AttackStrength.LOW) {
+            SLEEP_SHORT_TIME = 1000;
+            SLEEP_LONG_TIME = 2000;
+            doCrashScan = false;
+            doAllDataScan = true;
+            doTimedScan = true;
+            doJsonScan = true;
+            getMoreConfidence = false;
+            doAuthBypass = true;
+        } else if (this.getAttackStrength() == AttackStrength.MEDIUM) {
+            SLEEP_SHORT_TIME = 1000;
+            SLEEP_LONG_TIME = 3000;
+            doCrashScan = true;
+            doAllDataScan = true;
+            doTimedScan = true;
+            doJsonScan = true;
+            getMoreConfidence = true;
+
+            doAuthBypass = true;
+        } else if (this.getAttackStrength() == AttackStrength.INSANE) {
+            SLEEP_SHORT_TIME = 1000;
+            SLEEP_LONG_TIME = 3000;
+            doCrashScan = true;
+            doAllDataScan = true;
+            doTimedScan = true;
+            doJsonScan = true;
+            getMoreConfidence = true;
+            doAuthBypass = true;
+        }
+    }
+
+    @Override
+    public void scan(HttpMessage msg, NameValuePair originalParam) {
+        // TODO add TYPE_JSON control as soon as available
+        isJsonPayload = originalParam.getType() == NameValuePair.TYPE_POST_DATA;
```

Comment:
I believe this is checking if the parameter is in a POST body and not if the body is of JSON type.

Reply:
That's because the JSON variant used `TYPE_POST_DATA` (it was changed in zaproxy/zaproxy#4415).